pkg/csi/manila/controllerserver.go

```diff
@@ -135,9 +136,26 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 
 	sizeInGiB := bytesToGiB(requestedSize)
 
+	var accessibleTopology []*csi.Topology
+	accessibleTopologyReq := req.GetAccessibilityRequirements()
+	if cs.d.withTopology && accessibleTopologyReq != nil {
+		// All requisite/preferred topologies are considered valid. Nodes from those zones are required to be able to reach the storage.
+		// The operator is responsible for making sure that provided topology keys are valid and present on the nodes of the cluster.
+		accessibleTopology = accessibleTopologyReq.GetPreferred()
+
+		// Check if "availability" parameter is already specified, if not get preferred from Topology
+		// Required, incase share AZ is different from node AZ
+		if shareOpts.AvailabilityZone == "" {
+			shareOpts.AvailabilityZone = util.GetAZFromTopology(topologyKey, accessibleTopologyReq)
```

Comment:
Reviewers note: If AZ was unset in shareOpts, we now set it based on any topology in the request. This is new behaviour.